src/hierarchical_interleaved_bloom_filter.cpp

```diff
@@ -34,6 +34,16 @@
 namespace seqan::hibf
 {
 
+void hierarchical_interleaved_bloom_filter::set_number_of_user_bins()
+{
+    robin_hood::unordered_flat_set<int64_t> user_bins{};
+    for (auto const & ibf : ibf_bin_to_user_bin_id)
+        for (int64_t user_bin : ibf)
+            user_bins.emplace(user_bin);
+    user_bins.erase(-1); // Merged bin is no user bin
+    number_of_user_bins = user_bins.size();
```

Comment:
```suggestion
    int max_bin_id{-1};
    for (auto const & ibf : ibf_bin_to_user_bin_id)
        max_bin_id = std::max(max_bin_id, *std::ranges::max_element(ibf.begin(), ibf.end())); 
    assert(max_bin_id >= 0);
    number_of_user_bins = max_bin_id + 1;
```
no need to create a set I think?

Reply:
Good idea. `std::ranges::max_element` is the iterator interface, i.e. it returns an iterator. `std::ranges::max(ibf)` is easier :)

I removed the assert because the function should do nothing on an empty hibf (-1 + 1 = 0).